rest/model/event.go

```diff
@@ -0,0 +1,65 @@
+package model
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/evergreen-ci/evergreen/model/event"
+	"github.com/pkg/errors"
+)
+
+type APIEventLogEntry struct {
+	ID           *string        `bson:"_id" json:"-"`
+	ResourceType *string        `bson:"r_type,omitempty" json:"resource_type,omitempty"`
+	ProcessedAt  *time.Time     `bson:"processed_at" json:"processed_at"`
+	Timestamp    *time.Time     `bson:"ts" json:"timestamp"`
+	ResourceId   *string        `bson:"r_id" json:"resource_id"`
+	EventType    *string        `bson:"e_type" json:"event_type"`
+	Data         *TaskEventData `bson:"data" json:"data"`
+}
+
+type TaskEventData struct {
+	Execution int     `bson:"execution" json:"execution"`
+	HostId    *string `bson:"h_id,omitempty" json:"host_id,omitempty"`
+	UserId    *string `bson:"u_id,omitempty" json:"user_id,omitempty"`
+	Status    *string `bson:"s,omitempty" json:"status,omitempty"`
+	JiraIssue *string `bson:"jira,omitempty" json:"jira,omitempty"`
+
+	Timestamp *time.Time `bson:"ts,omitempty" json:"timestamp,omitempty"`
+	Priority  int64      `bson:"pri,omitempty" json:"priority,omitempty"`
+}
+
+func (el *APIEventLogEntry) BuildFromService(t interface{}) error {
+	switch v := t.(type) {
+	case *event.EventLogEntry:
+		d, ok := v.Data.(*event.TaskEventData)
+		if ok == false {
+			return errors.New(fmt.Sprintf("Incorrect type for data field when unmarshalling EventLogEntry"))
+		}
+		(*el) = APIEventLogEntry{
```

Comment:
I would instead do `el = &APIEventLogEntry{`

Reply:
making that change makes the Data field nil.

Reply:
I think that response was for the comment below? This is just a different way to write the same thing (but cleaner)

Reply:
Haha jk this one is new to me: https://stackoverflow.com/a/38105687
You can't change the identity of a data structure within itself I guess